x/mongo/driver/topology/fsm.go

```diff
@@ -35,8 +40,7 @@ func newFSM() *fsm {
 //
 // apply should operation on immutable descriptions so we don't have to lock for the entire time we're applying the
 // server description.
-func (f *fsm) apply(s description.Server) (description.Topology, description.Server, error) {
-
+func (f *fsm) apply(s description.Server) (description.Topology, description.Server) {
```

Comment:
Even though the FSM stores the compatiblity error, I'd pefer to have this function return the error as well so it's clear from the function signature that the caller has to check the error. Otherwise, it's up to the caller to remember that `fsm.compatibilityErr` has to be checked.